12-offer-encoding.md

```diff
@@ -0,0 +1,835 @@
+    - MUST set (or not set) `offer_metadata`, `offer_description`, `offer_absolute_expiry`, `offer_paths` and `offer_issuer` as it would for an offer.
+    - MUST NOT include `signature`, `offer_chains`, `offer_amount`, `offer_currency`, `offer_features`, `offer_quantity_min`, `offer_quantity_max` or `offer_node_id`
+    - if it sets `invoice_request_payer_id`:
+      - MUST set `key` to the node's public key to request the invoice from.
+    - otherwise:
+       - MUST provide at least one `offer_blinded_path`
+    - if the chain for the invoice is not solely bitcoin:
+      - MUST specify `invoice_request_chain` the offer is valid for.
+    - MUST set `invoice_request_amount`.
+  - MUST NOT set any tlv fields greater or equal to 160.
+  - MUST set `invoice_request_metadata` to an unpredictable series of bytes.
+  - if it sets `invoice_request_amount`:
+    - MUST set `msat` in multiples of the minimum lightning-payable unit
+        (e.g. milli-satoshis for bitcoin) for `invoice_request_chain` (or for bitcoin, if there is no `invoice_request_chain`).
+  - if it supports bolt12 invoice request features:
+    - MUST set `invoice_request_features`.`features` to the bitmap of features.
+
+The reader:
+  - MUST fail the request if `invoice_request_payer_id` is not present.
+  - MUST fail the request if any fields have type greater or equal to 160.
+  - if `invoice_request_features` contains unknown _odd_ bits that are non-zero:
+    - MUST ignore the bit.
+  - if `invoice_request_features` contains unknown _even_ bits that are non-zero:
+    - MUST fail the request.
+  - if `offer_node_id` is present (response to an offer):
+    - MUST fail the request if the offer fields do not exactly match a valid, unexpired offer.
+    - MUST fail the request if `invoice_request_signature` is not correct as detailed in [Signature Calculation](#signature-calculation) using the `invoice_request_payer_id`.
+    - if `offer_quantity_min` or `offer_quantity_max` is present:
+      - MUST fail the request if there is no `invoice_request_quantity` field.
+      - MUST fail the request if `invoice_request_quantity` is not within that (inclusive) range.
+    - otherwise:
+      - MUST fail the request if there is an `invoice_request_quantity` field.
+    - if `offer_amount` is present:
+      - MUST calculate the *base invoice amount* using the `offer_amount`:
+        - if `offer_currency` is not the `invoice_request_chain` currency, convert to the
+          `invoice_request_chain` currency.
+        - if `invoice_request_quantity` is present, multiply by `invoice_request_quantity`.`quantity`.
+      - if `invoice_request_amount` is present:
+        - MUST fail the request if `invoice_request_amount`.`msat` is less than the *base invoice amount*.
+        - MAY fail the request if `invoice_request_amount`.`msat` exceeds the *base invoice amount*.
+        - MUST use `invoice_request_amount`.`msat` as the *base invoice amount*.
```

Comment:
In the writer section, it isn't clear that the quantity must be accounted for when setting `invoice_request_amount`. In fact, it even says:

> - MUST specify `invoice_request_amount`.`msat` as greater or equal to amount expected by `offer_amount` (and, if present, `offer_currency`).

Which seems to imply the quantity _shouldn't_ be accounted for when setting `invoice_request_amount`. But here it implies that it is. Am I missing something?

Reply:
It should!  Fixed:

```
        - MUST specify `invoice_request_amount`.`msat` as greater or equal to amount expected by `offer_amount` (and, if present, `offer_currency` and `invoice_request_quantity`).
```